src/spikeinterface/core/sortinganalyzer.py

```diff
@@ -629,36 +630,89 @@ def set_temporary_recording(self, recording: BaseRecording):
             sparsity = ChannelSparsity(sparsity_mask, unit_ids, self.channel_ids)
+        elif self.sparsity is not None and units_to_merge is not None:
+            sparsity_mask = np.zeros((len(new_unit_ids), self.sparsity.mask.shape[1]), dtype=bool)
+            for unit_ind, unit_id in enumerate(new_unit_ids):
+                if unit_id in unit_ids:
+                    # This is a new unit, and the sparsity mask will be the intersection of the
+                    # ones of all merges
+                    id = np.flatnonzero(unit_ids == unit_id)[0]
+                    to_be_merged = units_to_merge[id]
+                    indices = self.sorting.ids_to_indices(to_be_merged)
+                    union_mask = np.sum(self.sparsity.mask[indices], axis=0) > 0
+                    intersection_mask = np.prod(self.sparsity.mask[indices], axis=0) > 0
+                    if merging_mode == "soft":
+                        thr = np.sum(intersection_mask) / np.sum(union_mask)
+                        assert (
+                            thr > sparsity_overlap
+                        ), f"A sparsity threshold of {thr} has been found, can not use soft mode"
+                        sparsity_mask[unit_ind] = intersection_mask
+                    elif merging_mode == "hard":
+                        sparsity_mask[unit_ind] = union_mask
+
+                else:
+                    # This means that the unit is already in the previous sorting
+                    index = self.sorting.id_to_index(unit_id)
+                    sparsity_mask[unit_ind] = self.sparsity.mask[index]
+            sparsity = ChannelSparsity(sparsity_mask, list(new_unit_ids), self.channel_ids)
         else:
             sparsity = None
 
         # Note that the sorting is a copy we need to go back to the orginal sorting (if available)
         sorting_provenance = self.get_sorting_provenance()
         if sorting_provenance is None:
-            # if the original sorting objetc is not available anymore (kilosort folder deleted, ....), take the copy
+            # if the original sorting object is not available anymore (kilosort folder deleted, ....), take the copy
             sorting_provenance = self.sorting
 
-        if unit_ids is not None:
+        if units_to_merge is None:
             # when only some unit_ids then the sorting must be sliced
             # TODO check that unit_ids are in same order otherwise many extension do handle it properly!!!!
             sorting_provenance = sorting_provenance.select_units(unit_ids)
+        else:
+            from spikeinterface.core.sorting_tools import apply_merges_to_sorting
+
+            sorting_provenance, keep_mask = apply_merges_to_sorting(
+                sorting=sorting_provenance,
+                units_to_merge=units_to_merge,
+                new_unit_ids=unit_ids,
+                censor_ms=censor_ms,
+                return_kept=True,
```

Comment:
we need the option of new_id_strategy here.

Reply:
No, because this is handled before, and new_unit_ids here is never None